File: JavaScript/Coding/Recursion/fib.js
function fibonacci(n) {
  // first two numbers ar 0 and 1
  var first = 0;
  var second = 1;

  if (n < 1) {
    return "n can't be less than 1";
  } else if (n === 1 || n === 2) {
    return n - 1;
  } else {
    return fibonacci(n - 1) + fibonacci(n - 2);
  }
}

function canConstruct(target, wordBank) {
  if (target == "") {
    return true;
  }
  for (let word of wordBank)
    if (target.indexOf(word) == 0) {
      suffix = target.slice(word.length);
      if (canConstruct(suffix, wordBank) == true) {
        return true;
      }
    }

  return false;
}
// 0 1 1 2 3
// console.log(fibonacci(5))
// console.log(fibonacci(10))
console.log(canConstruct("abcdef", ["ab", "abc", "cd", "def", "abcd"]));

File: Python_Programming/Divide_Conquer/fib.py
# O(2^n)
def fib(nth):
    # first two numbers are 0 and 1 as given
    if nth < 1:
        return "Number can't be less than 1"
    elif nth == 1:
        return 0
    elif nth == 2:
        return 1
    else:
        return fib(nth - 1) + fib(nth - 2)


print(fib(8))
